tutorials/vbpr_text.ipynb

```diff
@@ -0,0 +1,215 @@
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Cross modality\n",
+    "\n",
+    "To get representations from text data, we build a `TextModality` using text corpus with coressponding ids as the input. We also need to supply a `Tokenizer` for text splitting, in this case the tokens in text plots are seperated by `\\tab` character. We limit the maximum size of vocabulary to 5000, which also means the size of representations (e.g., bag-of-words vectors) can not go beyond that limit."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "item_text_modality = TextModality(corpus=plots, ids=movie_ids, \n",
+    "                                  tokenizer=BaseTokenizer(sep='\\t', stop_words='english'),\n",
+    "                                  max_vocab=5000, max_doc_freq=0.5).build()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After building the `TextModality`, we can access text representations through provided APIs and attributes of the object. In this case, we take the word-count matrix as item representations then use it as input to construct an `ImageModality`. In other words, item visual representations will be substituted by bag-of-words representations which are used by `VBPR` model. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "features = item_text_modality.count_matrix.A\n",
+    "item_image_modality = ImageModality(features=features, ids=movie_ids)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In Cornac, image recommendation models always work with `ImageModality`, similarly for other modalities. That design makes it consistent with models' original assumptions, thus it's easy to integrate models into the library without any confusion of which modality to use.\n",
+    "\n",
+    "## Experiment\n",
+    "\n",
+    "We employ `RatioSplit` evaluation method to split the rating data. The `item_image_modality` is also supplied here for later usage of model training."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ratio_split = RatioSplit(data=ml_100k, test_size=0.9,\n",
+    "                         item_image=item_image_modality,\n",
+    "                         exclude_unknowns=True, \n",
+    "                         verbose=True, seed=123)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "VBPR model is supported inside Cornac. Note that the implementation of VBPR uses [PyTorch](https://pytorch.org/), thus we need to install that dependency in order to run the model (done at the beginning). List of implemented models with their additional dependencies is carefully documented [here](https://github.com/PreferredAI/cornac#models).\n",
```

Comment:
It may be necessary to recall the dependency of VBPR on PyTorch. This is done in the Setup section, and there is a need for more details on this aspect, it may be better to add them in the latter section (Setup).

This paragraph can then be reduced to something like: " We are now ready to fit and evaluate VBPR. We further include the BPR model as a baseline to assess the effectiveness of the text auxiliary data (movie plots). Both models are already implemented in Cornac thereby making such an evaluation straightforward".

Reply:
noted